Makefile

```diff
@@ -1,123 +1,104 @@
-# Metadata about this makefile and position
-MKFILE_PATH := $(lastword $(MAKEFILE_LIST))
-CURRENT_DIR := $(patsubst %/,%,$(dir $(realpath $(MKFILE_PATH))))
+SHELL := /bin/bash -o pipefail
 
-# Ensure GOPATH
-GOPATH ?= $(HOME)/go
+# List of tests to run
+FILES ?= ./...
 
 # List all our actual files, excluding vendor
-GOFILES ?= $(shell go list $(FILES) | grep -v /vendor/)
+GOPKGS ?= $(shell go list $(FILES) | grep -v /vendor/)
+GOFILES ?= $(shell find . -name '*.go' | grep -v /vendor/)
 
 # Tags specific for building
 GOTAGS ?=
 
 # Number of procs to use
 GOMAXPROCS ?= 4
 
-PROJECT := $(CURRENT_DIR:$(GOPATH)/src/%=%)
-OWNER := $(notdir $(patsubst %/,%,$(dir $(PROJECT))))
-NAME := $(notdir $(PROJECT))
-EXTERNAL_TOOLS = \
-	github.com/golang/dep/cmd/dep
-
-# Current system information
-GOOS ?= $(shell go env GOOS)
-GOARCH ?= $(shell go env GOARCH)
-GOCACHE ?= $(shell go env GOCACHE)
-
-# List of tests to run
-FILES ?= ./...
+NAME := $(notdir $(shell pwd))
 
 # Test Service ID
 FASTLY_TEST_SERVICE_ID ?=
 
-# bootstrap installs the necessary go tools for development or build.
-bootstrap:
-	@echo "==> Bootstrapping ${PROJECT}"
-	@for t in ${EXTERNAL_TOOLS}; do \
-		echo "--> Installing $$t" ; \
-		go get -u "$$t"; \
-	done
-.PHONY: bootstrap
-
-clean:
-	@echo "==> Cleaning ${NAME}"
-	@rm -rf pkg
-.PHONY: clean
-
-# build builds the binary into pkg/
-build:
-	@echo "==> Building ${NAME} for ${GOOS}/${GOARCH}"
-	@env \
-		-i \
-		PATH="${PATH}" \
-		CGO_ENABLED="0" \
-		GOOS="${GOOS}" \
-		GOARCH="${GOARCH}" \
-		GOCACHE="${GOCACHE}" \
-		GOPATH="${GOPATH}" \
-		go build -a -o "pkg/${GOOS}_${GOARCH}/${NAME}" ${GOFILES}
-.PHONY: build
-
-# deps updates all dependencies for this project.
-deps:
-	@echo "==> Updating deps for ${PROJECT}"
-	@dep ensure -update
-	@dep prune
-.PHONY: deps
-
-# dev builds and installs the
-dev:
-	@echo "==> Installing ${NAME} for ${GOOS}/${GOARCH}"
-	@env \
-		-i \
-		PATH="${PATH}" \
-		CGO_ENABLED="0" \
-		GOOS="${GOOS}" \
-		GOARCH="${GOARCH}" \
-		GOCACHE="${GOCACHE}" \
-		GOPATH="${GOPATH}" \
-		go install ${GOFILES}
-.PHONY: dev
-
-# linux builds the linux binary
-linux:
-	@env \
-		GOOS="linux" \
-		GOARCH="amd64" \
-		$(MAKE) -f "${MKFILE_PATH}" build
-.PHONY: linux
+all: mod-download dev-dependencies tidy fmt fiximports test vet staticcheck ## Runs all of the required cleaning and verification targets.
```

Comment:
@phamann Should this run `test-race` instead? Or in addition?

Reply:
Unless we use locks, channels, goroutines, `-race` probably isn't going to identify anything and with our `test-race` make target there is an additional overhead of not running the tests in parallel.

Reply:
I do see that [we use a single mutex in a Request to the API](https://github.com/fastly/go-fastly/blob/9f19db2dee031b5db530fdfef9415dfb9ed02d41/fastly/client.go#L253-L254), but that's the only concurrency primitive in this project.

Reply:
My logic is a bit flawed. While probably unlikely, we could be making a call to a function that uses concurrency primitives and does so in an unsafe manner. 

It does protect against potential races being introduced. Therefore, in order to keep the iterative process of development as streamlined as possible, i'll just have `test-race` run in CI, not locally.